wiki0/src/client/components/tokenViewer.tsx

```diff
@@ -4,13 +4,44 @@ import { parseJwt } from '../utils';
 
 type TokenViewerProps = {
   token: string | undefined | null;
+  saml?: boolean;
 };
 
-function TokenViewer({ token }: TokenViewerProps) {
+const defaultProps = {
+  saml: false,
+};
+
+function TokenViewer({ token, saml }: TokenViewerProps) {
   if (!token) {
     return <div className="text-gray-400">No JWT token found</div>;
   }
 
+  const displayToken = () => {
+    let tokenJson = '';
+    if (saml) {
+      return (
+        <div className="max-w-md">
+          Tip: Copy and Paste into{' '}
+          <a
+            target="_blank"
+            href="https://samltool.io"
+            className="font-medium text-blue-600 dark:text-blue-500 hover:underline"
+            rel="noreferrer"
+          >
+            samltool.io
+          </a>
+          <pre className="">{atob(token)}</pre>
+        </div>
+      );
+    }
+    try {
+      tokenJson = JSON.stringify(parseJwt(token), null, 2);
+    } catch (e) {
+      tokenJson = 'Error parsing token';
+    }
+    return tokenJson;
```

Comment:
nit, can be cleaned up
```suggestion
    if (saml) {
      return (
        <div className="max-w-md">
          Tip: Copy and Paste into{' '}
          <a
            target="_blank"
            href="https://samltool.io"
            className="font-medium text-blue-600 dark:text-blue-500 hover:underline"
            rel="noreferrer"
          >
            samltool.io
          </a>
          <pre className="">{atob(token)}</pre>
        </div>
      );
    }
    try {
      return JSON.stringify(parseJwt(token), null, 2);
    } catch (e) {
      return 'Error parsing token';
    }
```